fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -135,8 +140,11 @@
 
     private static final PropertyConverter propertyConverter = new PropertyConverter();
 
-    private static Function<FedoraResource, Function<IdentifierConverter<Resource, FedoraResource>, Function<Boolean,
-            Stream<Triple>>>> getDefaultTriples = resource -> translator -> uncheck(minimal -> {
+    // A curried type accepting resource, translator, and "minimality", returning triples.
+    private static interface RdfGenerator extends Function<FedoraResource,
```

Comment:
I dunno if it would be useful outside the current context. It's not hard to take a pure type up into the API.  But it would be completely disconnected from any other type in the API, because fundamentally, this type is about impl. Hm.